### PR TITLE
apiextensions-apiserver: make API helpers reusable

### PR DESCRIPTION
**Pull request by sttts**
Needed for CRD openapi publishing

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by soltysh** [comment]
/assign @deads2k 
for approval

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go [hunk 1]
```diff
@@ -147,3 +149,108 @@ func IsStoredVersion(crd *CustomResourceDefinition, version string) bool {
 	}
 	return false
 }
+
+// GetSchemaForVersion returns the validation schema for given version in given CRD.
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go]: if we're exposing this, doc the possibility of returning `nil, nil` and when an error is returned (applies to all)

**Comment by liggitt** [comment]
/approve
/hold

one comment on doc. can address in follow up if desired

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go]: let me clean up the docs and also func names a bit, while we are at it.

Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go]: done

**Comment by sttts** [comment]
/hold cancel

Updated a couple of godocs.

**Comment by roycaihw** [comment]
LGTM

**Pull request merged**